apis/actions.md

```diff
@@ -0,0 +1,125 @@
+---
+description: Creating tool items, GLib.Actions, and keyboard shortcuts
+---
+
+# Actions
+
+GTK and GLib have a powerful API called [`GLib.Action`](https://valadoc.org/gio-2.0/GLib.Action.html) which can be used to define the primary actions of your app, assign them keyboard shortcuts, and tie them to [`Actionable`](https://valadoc.org/gtk+-3.0/Gtk.Actionable.html) widgets in your app like Buttons and Menu Items. In this section, we're going to create a Quit action for your app with an assigned keyboard shortcut and a Button that shows that shortcut in a tooltip. Begin by creating a `Gtk.Application` with a `Gtk.ApplicationWindow` as you've done in previous examples.
+
+## Gtk.HeaderBar
+
+Let's begin by creating a new [`Gtk.HeaderBar`](https://valadoc.org/gtk+-3.0/Gtk.HeaderBar.html). Typically your app will have a HeaderBar at the top of the window that will contain tool items, which users will interact with to trigger your app's actions.
+
+```vala
+protected override void activate () {
+    var headerbar = new Gtk.HeaderBar () {
+        show_close_button = true
+    };
+
+    var main_window = new Gtk.ApplicationWindow (this) {
+        default_height = 300,
+        default_width = 300,
+        title = "Actions"
+    };
+    main_window.set_titlebar (headerbar);
+    main_window.show_all ();
+}
+```
+
+Since we're using this HeaderBar as our app's main titlebar, we need to set `show_close_button` to `true` so that GTK knows to include window controls. We can then override our Window's built-in titlebar with the `set_titlebar ()` method.
+
+Now, still in the activate function, let's create a new [`Gtk.Button`](https://valadoc.org/gtk+-3.0/Gtk.Button.html) with a big colorful icon and add it to our headerbar:
+
+```vala
+protected override void activate () {
+    var button = new Gtk.Button.from_icon_name ("process-stop", Gtk.IconSize.LARGE_TOOLBAR);
+
+    var headerbar = new Gtk.HeaderBar () {
+        show_close_button = true
+    };
+    headerbar.add (button);
+
+    [...]
+}
+```
+
+elementary OS ships with a large set of system icons that you can use in your app for actions, status, and more. You can easily browse the full set using the app [LookBook](http://appcenter.elementary.io/com.github.danrabbit.lookbook/), available in AppCenter.
+
+If you compile your app, you can see that it now has a custom HeaderBar with a big red icon in it. But when you click on it, nothing happens.
+
+## GLib.SimpleAction
+
+Let's define a new Quit action by adding the following to the beginning of the `activate ()` method
+
+```vala
+var quit_action = new SimpleAction ("quit", null);
+
+add_action (quit_action);
+set_accels_for_action ("app.quit",  {"<Control>q", "<Control>w"});
+```
+
+and this to the end of the `activate ()` method:
+
+```vala
+quit_action.activate.connect (() => {
+    main_window.destroy ();
+});
+```
+
+You'll notice that we do a few things here:
```

Comment:
With the markdown editor I am using to view this, a blank line needed to be inserted here for the list to include the first line properly.